platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,493 @@
+message Roster {
+
+    /**
+     * Map of roster entries, one per consensus node.
+     * <p>
+     * This map SHALL contain roster entries in natural order of ascending node ids.
+     * The node Id key SHALL match the node Id in the value.
+     * This map SHALL NOT be empty.<br/>
+     */
+    map<uint64, RosterEntry> rosterEntries = 1;
+}
+```
+
+#### RosterEntry
+
+A roster entry will look as follows:
+
+```proto
+/**
+ * A single roster entry in the network state.
+ *
+ * Each roster entry in SHALL encapsulate the elements required
+ * to manage node participation in the Threshold Signature Scheme (TSS).<br/>
+ * All fields are REQUIRED.
+ */
+message RosterEntry {
+
+    /**
+     * A consensus node identifier.
+     * <p>
+     * Node identifiers SHALL be unique _within_ a shard and realm,
+     * but a node SHALL NOT, ever, serve multiple shards or realms,
+     * therefore the node identifier MAY be repeated _between_ shards and realms.
+     */
+    uint64 node_id = 1;
+
+    /**
+     * A consensus weight.
+     * <p>
+     * Each node SHALL have a weight of zero or more in consensus calculations.<br/>
+     * The consensus weight of a node SHALL be calculated based on the amount
+     * of HBAR staked to that node.<br/>
+     * Consensus SHALL be calculated based on agreement of greater than `2/3`
+     * of the total `weight` value of all nodes on the network.
+     */
+    uint64 weight = 2;
+
+    /**
+     * An RSA public certificate used for signing gossip events.
+     * <p>
+     * This value SHALL be a certificate of a type permitted for gossip
+     * signatures.<br/>
+     * This value SHALL be the DER encoding of the certificate presented.<br/>
+     * This field is REQUIRED and MUST NOT be empty.
+     */
+    bytes gossip_ca_certificate = 3;
+
+    /**
+     * An elliptic curve public signing key.<br/>
+     * This contains the _long term_ public key for this node.
+     * <p>
+     * This value SHALL be the DER encoding of the presented elliptic curve
+     * public key.<br/>
+     * This field is OPTIONAL (that is, it can initially be null)
+     * but once set, it can no longer be null.
+     */
+    bytes tss_encryption_key = 4;
+
+    /**
+     * A list of service endpoints for gossip.
+     * <p>
+     * These endpoints SHALL represent the published endpoints to which other
+     * consensus nodes may _gossip_ transactions.<br/>
+     * If the network configuration value `gossipFqdnRestricted` is set, then
+     * all endpoints in this list SHALL supply only IP address.<br/>
+     * If the network configuration value `gossipFqdnRestricted` is _not_ set,
+     * then endpoints in this list MAY supply either IP address or FQDN, but
+     * SHALL NOT supply both values for the same endpoint.<br/>
+     * This list SHALL NOT be empty.<br/>
+     * This list SHALL NOT contain more than `10` entries.<br/>
+     * The first two entries in this list SHALL be the endpoints published to
+     * all consensus nodes.<br/>
```

Comment:
I don't understand this. All nodes will know of all endpoints, since the roster is stored in state?

Reply:
Grandfathered comment from `AddressBook`'s proto. All of the invariants of the address book are supposed to apply here. @jsync-swirlds, thoughts?

Reply:
I agree with Richard, the roster is in the state, so all entries are visible to all parties.

Reply:
nuked.